test/unit/inbound/process-submit-payment-request.test.js

```diff
@@ -0,0 +1,349 @@
+    getPaymentRequestByInvoiceNumber.mockReset()
+    getPaymentRequestByInvoiceNumber.mockReturnValueOnce(paymentRequest)
+
+    await processSubmitPaymentRequest(paymentRequest)
+
+    expect(mockTransaction.rollback).toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.rollback once when a valid paymentRequest is given and a previous paymentRequest exist', async () => {
+    getPaymentRequestByInvoiceNumber.mockReset()
+    getPaymentRequestByInvoiceNumber.mockReturnValueOnce(paymentRequest)
+
+    await processSubmitPaymentRequest(paymentRequest)
+
+    expect(mockTransaction.rollback).toHaveBeenCalledTimes(1)
+  })
+
+  test('should throw when getPaymentRequestByInvoiceNumber throws', async () => {
+    getPaymentRequestByInvoiceNumber.mockReset()
+    getPaymentRequestByInvoiceNumber.mockRejectedValue(new Error('Database retrieval issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw Error when getPaymentRequestByInvoiceNumber throws Error', async () => {
+    getPaymentRequestByInvoiceNumber.mockReset()
+    getPaymentRequestByInvoiceNumber.mockRejectedValue(new Error('Database retrieval issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw error with "Database retrieval issue" when getPaymentRequestByInvoiceNumber throws error with "Database retrieval issue"', async () => {
+    getPaymentRequestByInvoiceNumber.mockReset()
+    getPaymentRequestByInvoiceNumber.mockRejectedValue(new Error('Database retrieval issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(/^Database retrieval issue$/)
+  })
+
+  test('should throw when saveInvoiceNumber throws', async () => {
+    saveInvoiceNumber.mockRejectedValue(new Error('Database save down issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw Error when saveInvoiceNumber throws Error', async () => {
+    saveInvoiceNumber.mockRejectedValue(new Error('Database save down issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw error with "Database save down issue" when saveInvoiceNumber throws error with "Database save down issue"', async () => {
+    saveInvoiceNumber.mockRejectedValue(new Error('Database save down issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(/^Database save down issue$/)

```

Comment:
I like this syntax.  It's a neater that what I've been doing previously for async exception checks.